BitFaster.Caching/Lru/CustomPolicy.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace BitFaster.Caching.Lru
+{
+    /// <summary>
+    /// Defines a policy for determining the expiry of an item.
+    /// </summary>
+    public interface IExpiry<K, V>
+    {
+        /// <summary>
+        /// Gets the expiry for an item after it is created.
+        /// </summary>
+        Func<K, V, TimeSpan> GetExpireAfterCreate { get; }

```

Comment:
This part shouldn't actually be a func, it should be a method on the interface.

That way, it is possible to implement this without delegates.